### PR TITLE
Fix CryptoRandomHex function

### PR DESCRIPTION
**Pull request by jimbishopp**
Fixes https://github.com/gravitational/teleport-private/issues/80
LAT-APP21-5: Insecure random number generation

- updated rand call from rand.Reader.Read to rand.Read

minor nits:
- changed length parameter name from len to l.
- changed byte slice var name to b.
- updated godoc

### CODE REVIEW

**Review by zmb3 — DISMISSED** [review]

File: lib/utils/rand.go
Comment (r0mant) [inline lib/utils/rand.go:31]: Nit: It's not a big deal here since the function is small but in general descriptive variable names are preferred over single-letter variables like `b` or `l`. Makes the code much easier to read and follow.

**Review by r0mant — DISMISSED** [review]

File: lib/utils/rand.go
Comment (jimbishopp) [reply lib/utils/rand.go:31]: I'll change it to length. Shadowing len is what caught my eye.

**Review by zmb3 — DISMISSED** [review]

**Review by r0mant — DISMISSED** [review]

**Comment by jimbishopp** [comment]
Sorry, I may not have needed to merge the master. It seemed like Github wanted me to. LMK what the process is after approvals and the master is out of date. Not sure if I need to rebase or squash merge. It seems like every change triggers a re-review. Sorry for the noise, I'll figure this out soon.

**Comment by r0mant** [comment]
@jimbishopp Yeah we use squash and merge strategy and you do need to rebase off of master prior to merging. I think bot resets reviews because you're treated as external contributor since you're not a part of the reviewers map. @russjones should merge https://github.com/gravitational/teleport/pull/9159 soon which will fix that logic. In the meanwhile, you can just enable auto-merge (make sure to use squash strategy).

**Comment by russjones** [comment]
@jimbishopp Should be fixed now and the bot should no longer dismiss your approvals. Sorry about that!

**Review by russjones — APPROVED** [review]

**Review by r0mant — APPROVED** [review]

**Comment by jimbishopp** [comment]
@r0mant @zmb3 I finally got drone to pass after rerunning multiple times. The checks say "Checking reviewers - waiting for status". I assume this means you must reapprove? Sorry for the multiple approvals on this. It took a while for CI to work.

**Comment by r0mant** [comment]
@russjones @quinqu Looks like the `Checking reviewers` check is stuck on this PR for some reason - do you know why?

**Review by r0mant — APPROVED** [review]

**Review by zmb3 — APPROVED** [review]

**Pull request merged**
